horovod/ray/runner.py

```diff
@@ -284,12 +285,22 @@ def __init__(self,
                  num_hosts: int = 1,
                  num_slots: int = 1,
                  cpus_per_slot: int = 1,
-                 use_gpu: bool = False):
+                 use_gpu: bool = False,
```

Comment:
What do you think about making the default for `use_gpu=None`?

The logic would then become:

```
RayExecutor(..., use_gpu=True)  -> 1 GPU per slot
RayExecutor(..., gpus_per_slot=2)  -> 2 GPUs per slot
RayExecutor(..., use_gpu=True, gpus_per_slot=2)  -> 2 GPUs per slot
RayExecutor(...)  -> 0 GPUs per slot
RayExecutor(..., use_gpu=False, gpus_per_slot=1)  -> ValueError
```

Feels intuitive to me, but not sure if it makes things more confusing.  Feel free to leave as is if you think the current approach makes more sense.

Reply:
Hmm interesting. I think I will keep it as is to minimize the number of ways to do the most common case (1 gpu per worker) - the proposal would give you 3 ways:
 
* use_gpu=True + gpus_per_slot=1, 
* use_gpu=None + gpus_per_slot=1, 
* use_gpu=True + gpus_per_slot=None

while the status quo is 2 ways:
* use_gpu=True + gpus_per_slot=1, 
* use_gpu=True + gpus_per_slot=None

Reply:
Sounds good!